src/main/java/bio/terra/service/load/LoadDao.java

```diff
@@ -189,6 +203,65 @@ public void setLoadFileFailed(UUID loadId, String targetPath, String error) {
         updateLoadFile(loadId, targetPath, BulkLoadFileState.FAILED, null, error, null);
     }
 
+    public BulkLoadResultModel makeBulkLoadResult(UUID loadId) {
+        final String bulkLoadResultSql = "SELECT state, count(*) AS statecount FROM load_file" +
+            " WHERE load_id = :load_id GROUP BY state";
+        MapSqlParameterSource params = new MapSqlParameterSource();
+        params.addValue("load_id", loadId);
+        // Note: the cast of `rs` is needed because ResultSetExtractor and RowCallbackHandler have duplicate signatures
+        return jdbcTemplate.query(bulkLoadResultSql, params,
+            (ResultSetExtractor<BulkLoadResultModel>) rs -> {
+                BulkLoadResultModel result = new BulkLoadResultModel()
+                    .succeededFiles(0)
+                    .failedFiles(0)
+                    .notTriedFiles(0)
+                    .totalFiles(0);
+
+                while (rs.next()) {
+                    BulkLoadFileState state = BulkLoadFileState.fromValue(rs.getString("state"));
+                    if (state == null) {
+                        throw new CorruptMetadataException("Invalid file state");
+                    }
+                    switch (state) {
+                        case RUNNING:
+                            throw new CorruptMetadataException("No loads should be running!");
+
+                        case FAILED:
+                            result.setFailedFiles(rs.getInt("statecount"));
+                            break;
+
+                        case NOT_TRIED:
+                            result.setNotTriedFiles(rs.getInt("statecount"));
+                            break;
+
+                        case SUCCEEDED:
+                            result.setSucceededFiles(rs.getInt("statecount"));
+                            break;
```

Comment:
Can `state` end up invalid? We should add a `default` case if so.

Reply:
It "should never happen", so I added a default case :)